Doc/library/typing.rst

```diff
@@ -1500,6 +1510,76 @@ These are not used in annotations. They are building blocks for declaring types.
    ``True`` as the value of the ``total`` argument. ``True`` is the default,
    and makes all items defined in the class body required.
 
+   It is possible for a ``TypedDict`` type to inherit from one or more TypedDict types
+   using the class-based syntax.
+   Usage::
+
+      class Point3D(Point2D):
+         z: int
+
+   Now the ``Point3D`` has three items: ``x``, ``y`` and ``z``. It is equivalent to this
+   definition::
+
+      class Point3D(TypedDict):
+         x: int
+         y: int
+         z: int
+
+   The type info for introspection can be accessed via ``Point2D.__annotations__``,
+   ``Point2D.__total__``, ``Point2D.__required_keys__``, and
+   ``Point2D.__optional_keys__``.
+
+   .. attribute:: __annotations__
+
+      ``Point2D.__annotations__`` gives a dict which maps the field names to the field types::
+
+         >>> from typing import TypedDict
+         >>> class Point2D(TypedDict):
+         ...     x: int
+         ...     y: int
+         ...
+         >>> Point2D.__annotations__
+         {'x': <class 'int'>, 'y': <class 'int'>}
+
+   .. attribute:: __total__
+
+      ``Point2D.__total__`` gives the value of the ``total`` argument. Furthermore,
+      inherited items won't be affected by the totality flag which defined in superclass,
+      and instead use totality of the TypedDict type where they were defined.
```

Comment:
The second sentence is hard to parse, and feels out of place for the documentation of the `__total__` argument. I suggest to just omit it.